qbit-http-storages/src/commonMain/kotlin/qbit.storage/YandexDiskStorage.kt

```diff
@@ -0,0 +1,250 @@
+    private fun getFileNamesInResourceByType(resource: Resource, type: String): List<String> {
+        val items = resource._embedded?.items
+        var files: List<String> = emptyList()
+        if (items != null) {
+            files = items.filter { item -> item.type == type }.map { item -> item.name }
+        }
+        return files
+    }
+
+    /**    Helper method for getting full path from namespace to root */
+    private fun getFullPath(namespace: Namespace): String {
+        var parts = namespace.parts
+        return parts.joinToString("/") + "/"
+    }
+
+    /**    Helper method for getting paths from namespace and its parents */
+    private fun generatePutResourcePaths(namespace: Namespace): List<String> {
+        val partsArrayList = arrayListOf<List<String>>()
+        val pathsArrayList = arrayListOf<String>()
+        var currentNamespace = namespace
+        while (currentNamespace.parent != null) {
+            partsArrayList.add(currentNamespace.parts)
+            currentNamespace = currentNamespace.parent!!
+        }
+
+        val iterator = partsArrayList.iterator();
+        while (iterator.hasNext()) {
+            val parts = iterator.next()
+            pathsArrayList.add(parts.joinToString("/") + "/")
+        }
+        return pathsArrayList.reversed()
+    }
+
+    /**    Helper method for getting full path to key, including key name */
+    private fun getFullPath(key: Key): String {
+        val fileName = key.name
+        val namespace = key.ns
+        val parts = namespace.parts
+        val filePath = parts.joinToString("/")
+        return "$filePath/$fileName"
+    }
+}
+
+@InternalCoroutinesApi
+fun <T> runBlocking(body: suspend () -> T): T {
+    var res: T? = null
+    val job = GlobalScope.launch {
+        res = body()
+    }
+    while (job.isActive) {
+    }
+    if (job.isCancelled) {
+        job.getCancellationException().let { throw it.cause ?: it }
+    }
+    return res!!
+}
+
+@Serializable
+data class Resource(
+    val antivirus_status: String? = null, val resource_id: String? = null, val share: ShareInfo? = null,
+    val file: String? = null, val size: Int? = null, val photoslice_time: String? = null,
+    val _embedded: ResourceList? = null, val exif: Exif? = null, val custom_properties: JsonObject? = null,
+    val media_type: String? = null, val preview: String? = null, val type: String,
+    val mime_type: String? = null, val revision: Long? = null, val public_url: String? = null, val path: String,
+    val md5: String? = null, val public_key: String? = null, val sha256: String? = null, val name: String,
+    val created: String, val modified: String, val comment_ids: CommentIds? = null
+)
+
+@Serializable
+data class ShareInfo(val is_root: Boolean? = null, val is_owned: Boolean? = null, val rights: String)
+
+@Serializable
+data class ResourceList(
+    val sort: String? = null, val items: Array<Resource>, val limit: Int? = null, val offset: Int? = null,
```

Comment:
Are you sure, that you have to keep items in Array? If yes, there are idea warning - fix it

Reply:
replaced by List